skycoin/skywire
# Implement Route Multiplexer

Did you run `make format && make check`? Not yet

Fixes #440

Changes:	
- Implement Route Multiplexer with backward compatibility 

How to test this PR:
- Run the generic integration env, it should set up the following transports: `A -> B`, `C -> B`
- Set up a `C -> A` transport (`./bin/skywire-cli --rpc $RPC_A visor add-tp $PK_C --type stcpr`)
- Send a message from chat C to chat A (`curl --data {'"recipient":"'$PK_A'", "message":"Hello"}' -X POST $CHAT_C`)
- The message should appear in `VisorA` logs; logs of sending should appear in `VisorC` logs, as well as a message what transport was used (it should be `C -> A` and a total amount of transports)
- Delete the `C -> A` transport (`./bin/skywire-cli --rpc $RPC_C visor rm-tp cad4c72c-90fa-0117-b225-41eca240e785`)
- Send the message again
- Message should appear in `VisorA`'s logs and be sent from `VisorC` using the 2nd transport via `VisorB`
